### PR TITLE
Esp32 arduino core 2.0.3 support

### PR DESCRIPTION
**Pull request by Jason2866**
fixes the compile error `esp32_gpioMux` for ESP32, ESP32-C3, ESP32-S2 and ESP32-S3

### CODE REVIEW

File: util/OneWire_direct_gpio.h [hunk 1]
```diff
@@ -127,112 +127,94 @@
 static inline __attribute__((always_inline))
 IO_REG_TYPE directRead(IO_REG_TYPE pin)
 {
+#if CONFIG_IDF_TARGET_ESP32C3
+    return (GPIO.in.val >> pin) & 0x1;
+#else // plain ESP32
     if ( pin < 32 )
         return (GPIO.in >> pin) & 0x1;
-    else if ( pin < 40 )
+    else if ( pin < 46 )
```
Comment (lbernstone) [inline util/OneWire_direct_gpio.h:135]: Can't this be digitalPinCanOutput(pin) to be a bit more discriminating for the variants?

Comment (Jason2866) [reply util/OneWire_direct_gpio.h:135]: Good question, probably. Did just the needed changes, without touching existing code.

Comment (lbernstone) [reply util/OneWire_direct_gpio.h:135]: Sorry, that line (for the input) should be digitalPinIsValid(). Later, in the output pin validation it should be digitalPinCanOutput().

Comment (PaulStoffregen) [reply util/OneWire_direct_gpio.h:135]: Please send a pull request

Comment (PaulStoffregen) [reply util/OneWire_direct_gpio.h:135]: @lbernstone - Are you planning to send a pull request to use digitalPinCanOutput()?  Or in other words, should I hold off making an official release to make this fix with number 46 hard coded available to the wide ESP32 Arduino community?

Comment (PaulStoffregen) [reply util/OneWire_direct_gpio.h:135]: At least a few people have now confirmed this solves the problems on ESP32.  So unless someone sends a pull request, and even more people step up to give feedback (definitely the rate limiting factor), I'm leaning towards making a release with this code people have tested and confirmed is working.

**Pull request merged**